File: JavaScript/Easy/MajorityElement.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
  var majEl = null;
  var count = 0;
  for (var i = 0; i < nums.length; i++) {
    if (count == 0) {
      majEl = nums[i];
      count++;
    } else if (majEl == nums[i]) {
      count++;
    } else {
      count--;
    }
  }
  return majEl;
};

File: Python/Easy/MajorityElement.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        maj_el = None
        count = 0

        for num in nums:
            if count == 0:
                maj_el = num
                count += 1
            elif maj_el == num:
                count += 1
            else:
                count -= 1

        return maj_el
